### PR TITLE
roc_audio: improve watchdog

### PR DESCRIPTION
**Pull request by dshil**
#115

### CODE REVIEW

File: src/modules/roc_audio/watchdog.cpp [hunk 1]
```diff
@@ -45,19 +52,102 @@ bool Watchdog::update(packet::timestamp_t time) {
...
+
+    return true;
+}
+
+void Watchdog::check_frame_empty_(const Frame& frame) {
+    if (frame.flags() & audio::Frame::FlagEmpty) {
+        return;
+    }
+
+    read_time_ = update_time_;
```
Comment (gavv) [inline src/modules/roc_audio/watchdog.cpp]: I guess we also need `first_ = false` here.

**Comment by gavv** [comment]
It took me about a year to merge this. Sorry for delay, thanks!

**Comment by gavv** [comment]
Follow-up refactoring and minor fixes: b2ee64ddb7d0df9d501925c790c3572d0080b805

**Comment by dshil** [comment]
Never mind, great to see that you've come back!

**Comment by gavv** [comment]
One more follow-up requst: e5c3153e88871e19ecfd993cbb9efc761327f526

**Comment by gavv** [comment]
And one more after some testing: 2adcb64603dc9520076360d250bb7e7fbc929f99

**Pull request merged**